src/integer.c

```diff
@@ -516,9 +514,7 @@ Obj ObjInt_Int( Int i )
 Obj ObjInt_UInt( UInt i )
 {
   Obj gmp;
-  UInt bound = 1UL << NR_SMALL_INT_BITS;
-
-  if (i < bound) {
+  if (i <= INT_INTOBJ_MAX) {
```

Comment:
This is relying on how C handles signed / unsigned comparisons. I can't remember his they come out, and would have a small preference for assigning INT_INTOBJ_MAX to a UInt, just in case future people worry.

Reply:
I briefly thought about adding an `UINT_INTOBJ_MAX`, but the C standard is pretty clear on this, and always has been. In a (simplified) nutshell, a signed and unsigned value are compared by casting the signed value to unsigned, which is always what we want for these bounds checks.

In short, `INT_INTOBJ_MAX` and `UINT_INTOBJ_MAX` would always do exactly the same thing, *except* when accidentally using `UINT_INTOBJ_MAX` to bound check a signed quantity... As such, `UINT_INTOBJ_MAX` could be misused by accident. I don't see a way to misuse `INT_INTOBJ_MAX` by accident.